android/src/main/java/com/reactlibrary/RNImapMailerModule.java

```diff
@@ -70,8 +70,10 @@ public void run() {
 
                 } catch (NoSuchProviderException e) {
                     e.printStackTrace();
+                    promise.reject(e.getMessage());
```

Comment:
I think we can aggregate multiple catch statements into a single clause and re-use the same logic